### DISCUSSION TITLE
Using ROS with pycrazyswarm

### DISCUSSION THREAD

**Discussion by COLVERTYETY** [USC-ACTLab/crazyswarm#586] [Q&A]
Hello, I am an engineering student that is currently working with pycrazyswarm and I have a couple of questions about how to use crazyswarm with ros:

#### How do you access private ros params in a python script when the node has been renamed from a launch file?
You cannot start a node since that clashes with the `pycrazyswarm.Crazyswarm()` call.  
I have been solving this problem on my side by manually parsing the `sys.argv` to retrieve the new node name.

#### How do you then pass the `--sim` arg to the crazyswarm node ?
I had to manually inject it in the `sys.argv` params that I pass to `pycrazyswarm.Crazyswarm()`.

#### How do you solve the ros python path issues? 
I have solved  this issue on my side by `pip` installing the local pycrazyswarm folder.
This improves quality of life for development since symbolic links are no longer required.
This doesn't break back compatibility.  

Would a helper lib/functions to assist with parsing the args and retrieving the params be an interesting contribution to this repo?
Maybe the pycrazyswarm lib should be pip installed by default? ( maybe make it available on pypi?)

Please met me know if I have been missing easy answers to my questions and if my ideas make sens.
Thank you.

**Comment by JeremieLGDP** [comment]
We've got the same issues here, I'd love to know the answers, thanks!

**Comment by whoenig** [comment]
I don't fully understand the problems. There seems to be one with "ROS params", and another one with "ROS path issues". Could you provide a small example for what you did, what behavior you got, and what behavior you want instead?

**Comment by COLVERTYETY** [comment]
if I try to use a launch file like this:

```
<launch>
    <node name="cf1" pkg="decenter" type="exemple3.py">
        <param name="cfid" value="1"/>
    </node>
    <node name="cf2" pkg="decenter" type="exemple3.py">
        <param name="cfid" value="2"/>
    </node>
</launch>

```

I'd like to read the private cfid param. However `rospy.get_param("~cfid")` won't work because that requires a `rospy.init_node("my node")` and that has a conflict with  `pycrazyswarm.Crazyswarm()`.



Trying to use `--sim` after the launch command doesn't work by default so you have to add the following lines to the launch file.
```
    <arg name="sim" default="False"/>
    <param name="sim" value="$(arg sim)"/>
```
And then you have to evaluate `rospy.get_param("/sim")` before calling `pycrazyswarm.Crazyswarm(args="--sim")`.


for me the following line that I found in the docs doesn't work: #[export PYTHONPATH=$PYTHONPATH:/path/to/crazyswarm/ros_ws/src/crazyswarm/scripts](https://crazyswarm.readthedocs.io/en/latest/howto/howto.html)

I solved it by installing the `pycrazyswarm` lib. This removes the need for symbolic links everywhere.

I Hope that this helps you understand my problems.
